cli/src/lib.rs

```diff
@@ -480,6 +482,34 @@ impl Iroha {
 
         Ok(handle)
     }
+
+    /// Spawns a task which subscribes on updates from configuration actor
+    /// and broadcasts them further to interested actors. This way, neither config actor nor other ones know
+    /// about each other, achieving loose coupling of code and system.
+    fn spawn_configuration_updates_broadcasting(
+        kiso: KisoHandle,
+        logger: LoggerHandle,
+    ) -> task::JoinHandle<()> {
+        tokio::spawn(async move {
+            let mut log_level_update = kiso
+                .subscribe_on_log_level()
+                .await
+                // FIXME: don't like neither the message nor inability to throw Result to the outside
+                .expect("Cannot proceed without working subscriptions");
+
+            loop {
+                tokio::select! {
+                    Ok(()) = log_level_update.changed() => {
+                        let value = *log_level_update.borrow_and_update();
+                        if let Err(error) = logger.reload_level(value).await {
+                            iroha_logger::error!("Failed to reload log level: {error}");
+                        };
+                    }
+                };
+                tokio::task::yield_now().await;
```

Comment:
I blindly followed the pattern in other `select!`s. It turned out that in other places loops need to yield because their futures are always awailable and it might block other actors from functioning. In this case, however, messages are expected to be rare, so I will remove it.

Original discussion: https://github.com/hyperledger/iroha/pull/3346#discussion_r1155532000

Reply:
Removed.